test/custom_class.ll

```diff
@@ -0,0 +1,512 @@
+; RUN: llvm-spirv %s -to-binary -o %t.spv
```

Comment:
Can this IR be reduced to the smallest subset of instructions which still reproduces the crash? There are a lot of SYCL library methods which are confusing.

Reply:
The problem is reproducible only with -mem2reg optimization.